### PR TITLE
RunAsGroup - Update DropDisabled[Alpha]Fields behaviour

### PR DESCRIPTION
**Pull request by ceshihao**
**What type of PR is this?**
> Uncomment only one, leave it on its own line:
>

/kind cleanup

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

ref #72169
/assign @liggitt

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
None
```

### CODE REVIEW

File: pkg/security/podsecuritypolicy/provider.go [hunk 1]
```diff
@@ -295,12 +295,8 @@ func (s *simpleProvider) ValidateContainer(pod *api.Pod, container *api.Containe
 	scPath := containerPath.Child("securityContext")
 	allErrs = append(allErrs, s.strategies.RunAsUserStrategy.Validate(scPath, pod, container, sc.RunAsNonRoot(), sc.RunAsUser())...)
 
-	if utilfeature.DefaultFeatureGate.Enabled(features.RunAsGroup) {
```
[the remark this answers is not in the document]
Comment (ceshihao) [reply pkg/security/podsecuritypolicy/provider.go]: Changed.

**Comment by liggitt** [comment]
change looks good, thanks. can you add a unit test for each type along the lines of https://github.com/kubernetes/kubernetes/commit/de96583d8cc3b8ab63b53d64f7af88430e862b0f#diff-be375bafa40dfabd8db5f4737d758ef1

**Comment by ceshihao** [comment]
add unit test for RunAsGroup in both pod and podsecuritypolicy.

@liggitt

**Pull request merged**